### PR TITLE
design-proposals: Clean template and add feature lifecycle section

### LINKED ISSUE #286: Enhance the design proposal template with a graduation-criteria section
The current design proposal template [1] is missing graduation criteria and feature lifecycle section.
While suggesting an enhancement or change, it would be useful to put focus on the planning in regards to the feature graduation.

The subject has been discussed as part of the feature lifecycle proposal [2] and its relevance to the proposal template has been raised in this [3] thread.

[1] https://github.com/kubevirt/community/blob/main/design-proposals/proposal-template.md
[3] https://github.com/kubevirt/community/pull/251/
[3] https://github.com/kubevirt/community/pull/251/files#r1489624264

**Describe the solution you'd like**:

It is suggested to update the design template with a new `graduation criteria` section that ref the feature-lifecycle and provides a plan for graduation.

**Describe alternatives you've considered**:
N/A

### PR DESCRIPTION
**Pull request by lyarwood**
/cc @xpivarc
/cc @vladikr
/cc @EdDev

**What this PR does / why we need it**:

As discussed at the unconference for v1.5.0 [1] we should start documenting the expected lifecycle of a given feature within our design proposals before later moving this tracking to an issue.

This change cleans up the template, removing multiple top level headings, adding whitespace around headings and adds a final section defining the lifecycle of a feature.

[1] https://unconference.kubevirt.io/D62SngbSRXO1uhTAc8Ikqg#xpivarc-Feature-lifecycleFuture-of-proposals---1-1-1-1

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #286

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: design-proposals/proposal-template.md [hunk 1]
```diff
@@ -1,43 +1,62 @@
...
 (an overview on the approaches used to functional test this design)
 
-# Implementation Phases
+## Implementation Phases
+
 (How/if this design will get broken up into multiple phases)
+
+## Feature lifecycle Phases
+
+(How and when will the feature progress through the Alpha, Beta and GA lifecycle phases)
```
Comment (iholder101) [inline design-proposals/proposal-template.md:62]: Awesome to see this!

In order to provide some more guidance, I'd add sub-sections for alpha, beta and GA.
```suggestion
## Feature lifecycle Phases

(How and when will the feature progress through the Alpha, Beta and GA lifecycle phases)

### Alpha:

### Beta:

### GA:
```

The Kubernetes KEP template does so with a [comment](https://github.com/kubernetes/enhancements/blob/f431a1af55437bbceb6b633daf246791050a5ddc/keps/NNNN-kep-template/README.md?plain=1#L366-L383):
```
#### Alpha

- Feature implemented behind a feature flag
- Initial e2e tests completed and enabled

#### Beta

- Gather feedback from developers and surveys
- Complete features A, B, C
- Additional tests are in Testgrid and linked in KEP

#### GA

- N examples of real-world usage
- N installs
- More rigorous forms of testing—e.g., downgrade tests and scalability tests
- Allowing time for feedback
```

WDYT?

**Comment by lyarwood** [comment]
/close

@iholder101 thanks for the review but I think folks want to push ahead with the new enhancements approach in v1.5.0 instead now so I'm going to close this out.

File: design-proposals/proposal-template.md (hunk 1 above)
Comment (EdDev) [reply design-proposals/proposal-template.md:62]: And it should be clear that the phases are not mandatory, e.g. a feature can reach GA directly. Usually if it is trivial.

**Review by EdDev — APPROVED** [review]
Thank you!

**Pull request closed without merging**